### PR TITLE
Adding strikethrough support via the `htmlStrike` highlight group

### PR DESCRIPTION
**Pull request by jcla1**
As was requested in #26 and implemented in #34 (but with an unsatisfactory highlight group) I've taken the liberty to alter that PR to use a better highlight group instead.

(I've also had to adjust @therod's changes to fit into the current state of the syntax file.)

### CODE REVIEW

**Comment by tomtomjhj** [comment]
I think the changes I made in #185 should be applied to this patch, since strikethrough is a variant of emphasis. See https://github.github.com/gfm/#strikethrough-extension-.

**Comment by jcla1** [comment]
Good catch, thanks @tomtomjhj! I've modified the syntax rule accordingly.

**Comment by tpope** [comment]
Let's go with `markdownStrike`. It avoids that big realignment and matches what HTML does.

File: syntax/markdown.vim [hunk 1]
```diff
@@ -180,6 +181,9 @@ hi def link markdownBold                  htmlBold
 hi def link markdownBoldDelimiter         markdownBold
 hi def link markdownBoldItalic            htmlBoldItalic
 hi def link markdownBoldItalicDelimiter   markdownBoldItalic
+hi def link markdownStrike                htmlStrike
+hi def link markdownStrikeDelimiter       htmlStrike
```
Comment (tpope) [inline syntax/markdown.vim:185]: Let's follow the established pattern:
```vim
hi def link markdownStrikeDelimiter       markdownStrike
```

**Pull request merged**